numba/cpython/randomimpl.py

```diff
@@ -1506,3 +1506,66 @@ def multinomial_impl(n, pvals, size=None):
                         "tuple or None, got %s" % (size,))
 
     return multinomial_impl
+
+# ------------------------------------------------------------------------
+# np.random.dirichlet
+
+@overload(np.random.dirichlet)
+def dirichlet(alpha, size=None):
+
+    dtype = np.float64
+
+    @register_jitable
+    def dirichlet_arr(alpha, out):
+
+        #Gamma distribution method to generate a Dirichlet distribution
+
+        a_len = len(alpha)
+        size = out.size
+        flat = out.flat
+        for i in range(0, size, a_len):
+            exp = np.zeros(len(alpha), dtype=np.float64)
+            #calculate gamma random numbers per alpha specifications
+            for k, w in enumerate(alpha):
+                exp[k] = np.random.gamma(w, 1)
+            #normalize sum to 1
+            exp /= exp.sum()
+            for j, v in enumerate(exp):
+                flat[i+j] = v.item()
+
+    if not isinstance(alpha, (types.Sequence, types.Array)):
+        raise TypeError("np.random.dirichlet(): alpha should be an "
+                        "array or sequence, got %s" % (alpha,))
+
+    if not isinstance(alpha, types.Array):
+        alpha = np.array(alpha)
+
+    if size in (None, types.none):
+        def dirichlet_impl(alpha, size=None):
+            out = np.zeros(len(alpha), dtype)
+            dirichlet_arr(alpha, out)
+            return out
+
+    elif isinstance(size, types.Integer):
+        def dirichlet_impl(alpha, size=None):
+            """
+            dirichlet(..., size=int)
+            """
+            out = np.zeros((size, len(alpha)), dtype)
+            dirichlet_arr(alpha, out)
+            return out
+
+    elif isinstance(size, types.BaseTuple):
```

Comment:
I'm not sure this is safe? What if the tuple is `('a', 1, 3j)` ?

Reply:
Good point. This line is also in np.random.multinomial - worth changing in this PR?